[DOCS] Add 'how to' doc about avoiding oversharding

kilfoyle opened the pull request:
Hi Elasticsearch writers! 👋 
This relates to an [issue](https://github.com/elastic/cloud/issues/55885) in the elastic/cloud docs, which suggests we should port in @bczifra's  really excellent article about [how to avoid oversharding](https://docs.google.com/document/d/1RGHEdJOxfk54TzA3tKSE0QDh6ERL4_fRBCPEVEYIOR4/edit#heading=h.lklidoi3jh6y). Since it's pertinent to Elasticsearch we thought it best to add here, and I can then link to the new content from the Cloud docs.

This is my first time pushing docs outside of the Cloud repos so I'd appreciate any help. In particular, I'm not sure about the back-porting process. The article seems applicable to all maintained versions of the ES docs.

Commit:
Add avoid oversharding how to doc

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[[avoiding-oversharding]]
```
```suggestion
[[avoid-oversharding]]
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[[avoiding-oversharding]]
+== Avoid oversharding
+
+Sometimes reducing the number of shards in a cluster while maintaining the same amount of data will lead to a more effective use of system resources (CPU, RAM, IO). In these situations, we consider the cluster “oversharded”. The number of shards where this inflection point occurs depends on a variety of factors, including available hardware, indexing load, data volume, the types of queries execut ...
```
Can you rewrap lines throughout to keep to a 80 character line length?

jrodewig replied:
> The number of shards where this inflection point occurs depends on a variety of factors, including available hardware, indexing load, data volume, the types of queries executed against the cluster, the rate of these queries being issued, and the volume of data being queried.

I think this would work better as an unordered list.

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+Sometimes reducing the number of shards in a cluster while maintaining the same amount of data will lead to a more effective use of system resources (CPU, RAM, IO). In these situations, we consider the cluster “oversharded”. The number of shards where this inflection point occurs depends on a variety of factors, including available hardware, indexing load, data volume, the types of queries execut ...
+
+[float]
+=== Why oversharding is inefficient
```
Where possible, we try to add explicit anchors. It ensure that links to the content are less likely to break.

We also try to use `[discrete]` over `[float]` as that's preferred in Asciidoctor. You'll still see lots of historical `[float]` attributes tho.
```suggestion
[discrete]
[[oversharding-inefficient]]
=== Why oversharding is inefficient
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+=== Why oversharding is inefficient
+
+Each segment has metadata that needs to be kept in (heap) memory. These include lists of fields, the number of documents, and terms dictionaries. As a shard grows in size, the size of its segments generally grow because smaller segments are <<index-modules-merge,merged>> into fewer, larger segments. This typically reduces the amount of heap required by a shard’s segment metadata for a given data  ...
```
```suggestion
Each segment has metadata that needs to be kept in heap memory. These include lists of fields, the number of documents, and terms dictionaries. As a shard grows in size, the size of its segments generally grow because smaller segments are <<index-modules-merge,merged>> into fewer, larger segments. This typically reduces the amount of heap required by a shard’s segment metadata for a given data volume. At a bare minimum shards should be at least larger than 1GB to make the most efficient use of memory. 
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+However, even though shards start to be more memory efficient at around 1GB, a cluster full of 1GB shards will likely still perform poorly. This is because having many small shards can also have a negative impact on search and indexing operations. Each query or indexing operation is executed in a single thread per shard of indices being queried or indexed to. The node receiving a request from a c ...
+
+[float]
+=== How to reduce shard counts and increase shard size
```
```suggestion
[discrete]
[[reduce-shard-counts-increase-shard-size]]
=== How to reduce shard counts and increase shard size
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+=== How to reduce shard counts and increase shard size
+
+Try these six methods to reduce oversharding.
```
I don't think the numbering is needed or helps here. These steps could be applied in any order. It will also be easier to add/remove tips without numbers.
```suggestion
Try these methods to reduce oversharding.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+Try these six methods to reduce oversharding.
+
+[float]
+==== 1. Reduce the number of shards for new indices
```
I don't think `[discrete]` or `[float]` is needed for H4s.
```suggestion
[[reduce-shards-for-new-indices]]
==== Reduce the number of shards for new indices
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+The <<rollover-index-api-example,rollover request>> body can also override that same `index.number_of_shards` setting.
+
+[float]
+==== 2. Create larger shards by increasing the rollover thresholds
```
```suggestion
[[create-larger-shards-by-increasing-rollover-thresholds]]
==== Create larger shards by increasing rollover thresholds
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+Take special note of any indices that are empty. These may be managed by an ILM policy that is rolling over the indices because the `max_age` threshold is met. In this case, you may need to adjust the policy to make use of the `max_docs` or `max_size` properties to prevent the creation of these empty indices. One example where this may happen is if one or more Beats stop sending data. If the ILM- ...
+
+
+[float]
+==== 3. Create larger shards by using index patterns spanning longer time periods
```
```suggestion
[[create-larger-shards-with-index-patterns]]
==== Create larger shards by using index patterns spanning longer time periods
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+If creating indices using Logstash, the {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index[index] property of the Elasticsearch output can be modified to a <<date-math-index-names,date math expression>> covering a longer time period. For example, use `logstash-%{+YYYY.MM}`` instead of `logstash-%{+YYYY.MM.dd}`` to create monthly, rather than daily, indices. Beat ...
+
+
+[float]
+==== 4. Shrink an existing index to fewer shards
```
```suggestion
[[shrink-existing-index-to-fewer-shards]]
==== 4. Shrink an existing index to fewer shards
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+{ref}/index-lifecycle-management.html[Index Lifecycle Management] also has a shrink action available as part of its warm phase.
+
+
+[float]
+==== 5. Reindex an existing index to fewer shards
```
```suggestion
[[reindex-an-existing-index-to-fewer-shards]]
==== 5. Reindex an existing index to fewer shards
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+The <<docs-reindex,Reindex API>> can be used to reindex from an existing index to a new index with fewer shards. After the data has been reindexed, the initial oversharded index can be deleted.
+
+[float]
+==== 6. Reindex indices from shorter periods into longer periods
```
```suggestion
[[reindex-indices-from-shorter-periods-into-longer-periods]]
==== Reindex indices from shorter periods into longer periods
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+}
+--------------------------------------------------
+
+link:https://www.elastic.co/guide/en/elasticsearch/client/curator/current/index.html[Curator] can also link:https://www.elastic.co/guide/en/elasticsearch/client/curator/current/ex_reindex.html#_filter_selected_indices_2[help with reindexing time-based indices].
```
Curator is outdated. I'd just remove this reference.
```suggestion
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+==== 6. Reindex indices from shorter periods into longer periods
+
+The <<docs-reindex,Reindex API>> can be used to reindex multiple small indices covering shorter time periods into a larger index covering a longer time period. For example, daily indices from October with naming patterns such as `foo-2019.10.11` could be combined into a monthly `foo-2019.10` index, like this:
```
```suggestion
You can use the <<docs-reindex,reindex API>> to reindex multiple small indices covering shorter time periods into a larger index covering a longer time period. For example, daily indices from October with naming patterns such as `foo-2019.10.11` could be combined into a monthly `foo-2019.10` index, like this:
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+==== 5. Reindex an existing index to fewer shards
+
+The <<docs-reindex,Reindex API>> can be used to reindex from an existing index to a new index with fewer shards. After the data has been reindexed, the initial oversharded index can be deleted.
```
```suggestion
You can use the <<docs-reindex,reindex API>> to reindex from an existing index to a new index with fewer shards. After the data has been reindexed, the oversharded index can be deleted.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+The {ref}/indices-shrink-index.html[Shrink index API] can shrink an existing index down to a fewer shards.
+
+{ref}/index-lifecycle-management.html[Index Lifecycle Management] also has a shrink action available as part of its warm phase.
```
```suggestion
<<index-lifecycle-management,{ilm}>> also has a <<ilm-shrink-action,shrink action>> available for indices in the warm phase.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+==== 4. Shrink an existing index to fewer shards
+
+The {ref}/indices-shrink-index.html[Shrink index API] can shrink an existing index down to a fewer shards.
```
```suggestion
You can use the <<indices-shrink-index,shrink index API>> to shrink an existing index down to a fewer shards.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+Creating indices covering longer time periods will also reduce the index, and thereby shard, counts and increase index sizes. For example, instead of daily indices, create monthly, or even yearly indices.
+
+If creating indices using Logstash, the {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index[index] property of the Elasticsearch output can be modified to a <<date-math-index-names,date math expression>> covering a longer time period. For example, use `logstash-%{+YYYY.MM}`` instead of `logstash-%{+YYYY.MM.dd}`` to create monthly, rather than daily, indices. Beat ...
```
```suggestion
If creating indices using {ls}, the {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-index[index] property of the {es} output can be modified to a <<date-math-index-names,date math expression>> covering a longer time period. For example, use `logstash-%{+YYYY.MM}`` instead of `logstash-%{+YYYY.MM.dd}`` to create monthly, rather than daily, indices. {beats} also lets you change the date math expression defined in the `index` property of the {es} output, such as for {filebeat-ref}/elasticsearch-output.html#index-option-es[Filebeat].
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+==== 3. Create larger shards by using index patterns spanning longer time periods
+
+Creating indices covering longer time periods will also reduce the index, and thereby shard, counts and increase index sizes. For example, instead of daily indices, create monthly, or even yearly indices.
```
```suggestion
Creating indices covering longer time periods reduces index and shard counts while increasing index sizes. For example, instead of daily indices, you can create monthly, or even yearly indices.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+If indices are rolled over by either directly using the Rollover API or by specifying the Index Lifecycle Management {ref}/_actions.html#ilm-rollover-action[Rollover action], then the rollover condition thresholds (`max_age`, `max_docs`, `max_size`) can be increased to allow the indices to grow to a larger size before being rolled over, thereby generating larger shards.
+
+Take special note of any indices that are empty. These may be managed by an ILM policy that is rolling over the indices because the `max_age` threshold is met. In this case, you may need to adjust the policy to make use of the `max_docs` or `max_size` properties to prevent the creation of these empty indices. One example where this may happen is if one or more Beats stop sending data. If the ILM- ...
```
```suggestion
Take special note of any empty indices. These may be managed by an {ilm-init} policy that is rolling over the indices because the `max_age` threshold is met. In this case, you may need to adjust the policy to make use of the `max_docs` or `max_size` properties to prevent the creation of these empty indices. One example where this may happen is if one or more {beats} stop sending data. If the {ilm-init}-managed indices for those {beats{ are configured to rollover daily, then new, empty indices will be generated each day. Empty indices can be identified using the <<cat-count,cat count API>>.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+==== 2. Create larger shards by increasing the rollover thresholds
+
+If indices are rolled over by either directly using the Rollover API or by specifying the Index Lifecycle Management {ref}/_actions.html#ilm-rollover-action[Rollover action], then the rollover condition thresholds (`max_age`, `max_docs`, `max_size`) can be increased to allow the indices to grow to a larger size before being rolled over, thereby generating larger shards.
```
```suggestion
You can roll over indices using the <<indices-rollover-index,rollover index API>> or by specifying the <<ilm-rollover-action,rollover action>> in an {ilm-init} policy. If using an {ilm-init} policy, increase the rollover condition thresholds (`max_age`, `max_docs`, `max_size`)  to allow the indices to grow to a larger size before being rolled over, which creates larger shards.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[float]
+==== 1. Reduce the number of shards for new indices
+
+The `index.number_of_shards` setting can be specified for new indices created with the <<indices-create-index,Create Index API>> or as part of <<indices-templates,index templates>> for indices automatically created by <<index-lifecycle-management,Index Lifecycle Management>>. For Elasticsearch versions below 7.0, the default value for this setting was `5`. For versions 7.0 and higher, the default ...
```
I don't think we need to call out the default values here.
```suggestion
You can specify the `index.number_of_shards` setting  for new indices created with the <<indices-create-index,create index API>> or as part of <<indices-templates,index templates>> for indices automatically created by <<index-lifecycle-management,{ilm} ({ilm-init})>>.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+The `index.number_of_shards` setting can be specified for new indices created with the <<indices-create-index,Create Index API>> or as part of <<indices-templates,index templates>> for indices automatically created by <<index-lifecycle-management,Index Lifecycle Management>>. For Elasticsearch versions below 7.0, the default value for this setting was `5`. For versions 7.0 and higher, the default ...
+
+The <<rollover-index-api-example,rollover request>> body can also override that same `index.number_of_shards` setting.
```
```suggestion
You can override the `index.number_of_shards`  when rolling over an index using the <<rollover-index-api-example,rollover index API>>.
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+
+[source,console]
+--------------------------------------------------
+POST _reindex
```
```suggestion
POST /_reindex
```

jrodewig commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,73 @@
+[[avoiding-oversharding]]
+== Avoid oversharding
+
+Sometimes reducing the number of shards in a cluster while maintaining the same amount of data will lead to a more effective use of system resources (CPU, RAM, IO). In these situations, we consider the cluster “oversharded”. The number of shards where this inflection point occurs depends on a variety of factors, including available hardware, indexing load, data volume, the types of queries execut ...
```
```suggestion
Sometimes reducing the number of shards in a cluster while maintaining the same amount of data will lead to a more effective use of system resources (CPU, RAM, IO). In these situations, we consider the cluster “oversharded”. The number of shards where this inflection point occurs depends on a variety of factors, including available hardware, indexing load, data volume, the types of queries executed against the cluster, the rate of these queries being issued, and the volume of data being queried. Testing against production data with production queries on production hardware is the only way to calibrate optimal shard sizes. Shard sizes of tens of GB are commonly used, and this may be a useful starting point from which to experiment. {kib}'s {kibana-ref}/elasticsearch-metrics.html[{es} monitoring] provides a useful view of historical cluster performance when evaluating the impact of different shard sizes.
```

jrodewig commented:
Thanks @kilfoyle! This looks good to me. I left several comments to update the syntax and wording. However, I'll defer to @debadair for a final review.

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: James Rodewig <james.rodewig@elastic.co>

Commit:
Address James' suggestions

kilfoyle commented:
@jrodewig  Thank you for the really excellent review!

I've incorporated all of your suggestions, including re-wrapping the text to within 80 characters (with the exception of one link at line 99 that spans further than that - hopefully that's okay). Also, I get warnings if the `[discrete]` tag isn't above the level four sections, so I added the tags back in. If that's a potential problem just let me know.

debadair commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,148 @@
+==== Shrink an existing index to fewer shards
+
+You can use the <<indices-shrink-index,shrink index API>> to shrink an
+existing index down to a fewer shards.
```
```suggestion
existing index down to fewer shards.
```

debadair commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,148 @@
+In this case, you may need to adjust the policy to make use of the `max_docs`
+or `max_size` properties to prevent the creation of these empty indices. One
+example where this may happen is if one or more {beats} stop sending data. If
+the {ilm-init}-managed indices for those {beats{ are configured to rollover
```
```suggestion
the {ilm-init}-managed indices for those {beats} are configured to roll over
```

debadair commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,148 @@
+[[avoid-oversharding]]
+== Avoid oversharding
+
+Sometimes reducing the number of shards in a cluster while maintaining the
+same amount of data will lead to a more effective use of system resources
```
```suggestion
In some cases, reducing the number of shards in a cluster while maintaining the
same amount of data leads to a more effective use of system resources
```

debadair commented on docs/reference/how-to/avoid-oversharding.asciidoc:
```diff
@@ -0,0 +1,148 @@
+
+Sometimes reducing the number of shards in a cluster while maintaining the
+same amount of data will lead to a more effective use of system resources
+(CPU, RAM, IO). In these situations, we consider the cluster “oversharded”.
```
```suggestion
(CPU, RAM, IO). In these situations, we consider the cluster _oversharded_.
```

debadair reviewed (changes requested):
Thanks for doing this @kilfoyle! Spotted a couple minor things that should be fixed, but otherwise LGTM. The backporting guidelines are in the docs readme: https://github.com/elastic/elasticsearch/blob/master/docs/README.asciidoc#backporting-doc-fixes.

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: debadair <debadair@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: debadair <debadair@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: debadair <debadair@elastic.co>

kilfoyle committed:
Update docs/reference/how-to/avoid-oversharding.asciidoc

Co-Authored-By: debadair <debadair@elastic.co>

kilfoyle commented:
Thanks @debadair for the very careful review!  I've committed your suggestions.

Forgive me for a dumb question, but I don't understand the backporting process.  From the guidelines it appears that this would need to go into releases from `7.0` up to `7.6`.  If either of you have some time could you perhaps help me with that part?

kilfoyle commented:
Awesome. Thanks a lot @jrodewig! Much appreciated.

debadair reviewed (approved):
LGTM

elasticmachine commented:
Pinging @elastic/es-distributed (:Distributed/Distributed)

jrodewig commented:
### Backport commits
master 8b4f60fd48c0a03e454beef3f5d6836067962b94
7.x 6f95139 
7.7 55d9802
7.6 a096492
